awx/main/access.py

```diff
@@ -1997,7 +1997,11 @@ class WorkflowJobNodeAccess(BaseAccess):
     )
 
     def filtered_queryset(self):
-        return self.model.objects.filter(workflow_job__unified_job_template__in=UnifiedJobTemplate.accessible_pk_qs(self.user, 'read_role'))
+        return self.model.objects.filter(
+            Q(workflow_job__unified_job_template__in=UnifiedJobTemplate.accessible_pk_qs(self.user, 'read_role'))
+            | Q(workflow_job__created_by_id=self.user.id, workflow_job__is_bulk_job=True)
+            | Q(workflow_job__organization__in=Organization.objects.filter(Q(admin_role__members=self.user)), workflow_job__is_bulk_job=True)
```

Comment:
As you can imagine, this is going to be a problem along with the WorkflowJob filter. But let me try to open up some options here.

Ideally, I should never care about this filter. It should be rarely used in normal operation. This is the only option to build the API list view of `/api/v2/workflow_job_nodes/`, and will remain that way. However, I see no reason the UI should ever use that. No one should ever need or want a global list of workflow job nodes. Instead, they should use the `/api/v2/workflow_job/42/workflow_nodes/` sub-list. Right now, this still uses this filter (actually worse, it uses this filter with JOIN and DISTINCT to do the parent filter).

It should be uncontroversial and easy to change `get_queryset` on `WorkflowJobWorkflowNodesList` and any similar views to completely bypass this filter. This is because permission is dictated 100% by the parent object, which is the workflow job. As you can see here, all filters flow through `workflow_job__`. So even if this grows to unworkable complexity, this particular filter may not necessarily become a feature blocker if we can make some mitigating changes.

Reply:
we addressed changing the WorkflowJobWorkflowNodesList view  in https://github.com/ansible/awx/pull/13462/commits/03a9031f81e37f5e9c87e7a01835d345a9a059f8